modules/common/src/main/scala/emil/Disposition.scala

```diff
@@ -1,22 +1,25 @@
 package emil
 
-sealed trait Disposition {def name: String}
+import cats.Hash
 
-/**
-  * Enumeration of disposition types for a Content-Disposition part header.
-  */
+sealed trait Disposition { self: Product =>
+  final def name: String = productPrefix.toLowerCase
+}
+
+/** Enumeration of disposition types for a Content-Disposition part header. */
 object Disposition {
-  case object Inline extends Disposition {val name = "inline"}
-  case object Attachment extends Disposition {val name = "attachment"}
+  case object Inline extends Disposition
+  case object Attachment extends Disposition
 
-  val values = Seq(Inline, Attachment)
+  def fromString(str: String): Either[String, Disposition] =
```

Comment:
Ah you could have kept your impl, it was nicer :) But doesn't matter :)

Reply:
I couldn't decide, then I made it the way it already is. That seemed more consistent to me. ;-)